### PR TITLE
Document Windows app lifecycle breaking change

### PR DESCRIPTION
**Pull request by yaakovschectman**
Add pages for the breaking change of the engine external windows must call to be included in app lifecycle logic.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [c] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/platform-integration/windows/extern_win.md
Comment (loic-sharma) [inline src/platform-integration/windows/extern_win.md:15]:
```suggestion
An example procedure can be found on the [documentation of this breaking change.][]

```

File: src/platform-integration/windows/extern_win.md
Comment (loic-sharma) [inline src/platform-integration/windows/extern_win.md:13]: Could we mention the Flutter framework APIs that would be incorrect if you don't follow this? Perhaps [`WidgetsBindingObserver.didChangeAppLifecycleState`](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver/didChangeAppLifecycleState.html)

I'm thinking this will help folks find this page by Googling `Windows broken didChangeAppLifecycleState`.

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (loic-sharma) [inline src/release/breaking-changes/win_lifecycle_process_function.md:9]:
```suggestion
external windows added to a Flutter app must invoke

```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: src/release/breaking-changes/index.md
Comment (loic-sharma) [inline src/release/breaking-changes/index.md:40]: Ideally this line would make it a little clearer who is affected by this break: Windows app that have non-Flutter windows. What do you think of a title like `[Windows] External windows should notify Flutter of top-level window messages`? I'm not 100% happy with that either, so feel free to use something else :)

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (loic-sharma) [inline src/release/breaking-changes/win_lifecycle_process_function.md:29]: What's the `CALLBACK` here? Should we remove that?

Comment (yaakovschectman) [reply src/release/breaking-changes/win_lifecycle_process_function.md:29]: It's a macro that defines the `__stdcall` calling convention for Win32 callbacks.

File: src/platform-integration/windows/extern_win.md
Comment (loic-sharma) [inline src/platform-integration/windows/extern_win.md:16]: I would consider moving this up to the first paragraph where you explain why this migration is necessary.

Perhaps something like:

```
When adding a non-Flutter window to a Flutter Windows app, it will not be part
of the logic for application lifecycle state updates by default. For example,
this means that when the external window is shown or hidden, the app lifecycle
state will not appropriately update to inactive or hidden. As a result,
the app may receive incorrect lifecycle state changes through 
[WidgetsBindingObserver.didChangeAppLifecycle][].

In order to add the
external window to this application logic, the window's `WndProc` procedure
must invoke `FlutterEngine::ProcessExternalWindowMessage`.
```

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (loic-sharma) [inline src/release/breaking-changes/win_lifecycle_process_function.md:30]: `FlutterEngine::ProcessExternalWindowMessage` isn't a static method, right? Most apps probably have a `FlutterViewController` member in their Flutter window ([example](https://github.com/flutter/flutter/blob/5649161bd7fc591e313c12f35fccfe4c8e211dbe/examples/hello_world/windows/runner/flutter_window.h#L34C28-L34C49)) and should so something like:

```
std::optional<LRESULT> result = flutter_controller_->engine()->ProcessExternalWindowMessage(hwnd, msg, wparam, lparma);
```

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (loic-sharma) [reply src/release/breaking-changes/win_lifecycle_process_function.md:29]: Oh derp that's a win32 thing, gotcha: https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types

File: src/platform-integration/windows/extern_win.md
Comment (cbracken) [reply src/platform-integration/windows/extern_win.md:13]: I would be tempted to format it something like this:

# Title

Single line very brief TL;DR style summary of the change.

## Who is affected
All Flutter app developers who maintain a Windows version of their app created that was created using a Flutter version earlier than X.

## What has changed
Details paragraph summarising what changed and how the new world differs from the old.

## What you need to do
Step by step instructions on how to migrate.

## Why this change was necessary
Background on why we needed to make this change.

Comment (cbracken) [reply src/platform-integration/windows/extern_win.md:13]: And reading down further... I could have saved myself some writing 🤣

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (cbracken) [inline src/release/breaking-changes/win_lifecycle_process_function.md:11]: I would be tempted to add a "Who is affected" section here.

File: src/platform-integration/windows/extern_win.md
Comment (cbracken) [inline src/platform-integration/windows/extern_win.md:7]: I'd be tempted to break out sections:

## Who is affected
Windows applications built against Fluter version X or later that open non-Flutter windows.

**Review by cbracken — COMMENTED** [review]
Overall the summary looks pretty good. Just a few suggestions on how to make it easy for readers to look at the page and know whether they need to care or not in 10 seconds or less, and general suggestion to make the migration instructions very step-by-step and point form rather than a paragraph.

**Review by cbracken — APPROVED** [review]
Emphatic lgtm!

LGTM stamp from a Japanese personal seal

**Comment by atsansone** [comment]
I'll get to this today.

**Comment by cbracken** [comment]
How's this looking @atsansone @yaakovschectman ?

File: src/release/breaking-changes/win_lifecycle_process_function.md
Comment (domesticmouse) [reply src/release/breaking-changes/win_lifecycle_process_function.md:11]: Please add this section

**Review by domesticmouse — APPROVED** [review]
LGTM once the Who is affected section is added.

PTAL @atsansone

**Comment by domesticmouse** [comment]
There is an invalid link in `release/breaking-changes/index.html` that needs to be fixed before this can land. See the CI failure for more detail.

**Comment by yaakovschectman** [comment]
@domesticmouse I cannot actually make sense of this CI failure. The error log states that the broken link is `[Windows build path changed to add the target architecture][]`, but that link is not added or even touched by this PR.

**Comment by domesticmouse** [comment]
PTAL @atsansone 

> @domesticmouse I cannot actually make sense of this CI failure. The error log states that the broken link is `[Windows build path changed to add the target architecture][]`, but that link is not added or even touched by this PR.

**Comment by domesticmouse** [comment]
> @domesticmouse I cannot actually make sense of this CI failure. The error log states that the broken link is `[Windows build path changed to add the target architecture][]`, but that link is not added or even touched by this PR.

I suspect what is happening is that in the link block that the `Windows build path changed to add the target architecture` link declaration is being broken by the link you are adding directly above it. The thing to remember is that markdown is a pile of regexs, stacked atop one another. I suspect having `[Windows]` as part of your link text is breaking the regex. Because there is no such thing as a MarkDown AST.

**Review by loic-sharma — APPROVED** [review]
LGTM from a technical standpoint

File: src/platform-integration/windows/extern_win.md
Comment (atsansone) [inline src/platform-integration/windows/extern_win.md:10]: **nitpick:** Don't use symbols. Use words.
```suggestion
Windows applications built against Flutter versions after 3.13 that open non-Flutter windows.

```

File: src/platform-integration/windows/extern_win.md
Comment (atsansone) [inline src/platform-integration/windows/extern_win.md:26]: **chore (non-blocking):** Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure). Write in active voice per [GSG](https://developers.google.com/style/voice).

```suggestion
To achieve this, add the following code to a window message handler function:
```

**Review by atsansone — APPROVED** [review]
LGTM % a few chores.

**Pull request merged**
